### DISCUSSION TITLE
Cannot read file/directory from getServerSideProps

### DISCUSSION THREAD

**Discussion by robertcoopercode** [vercel/next.js#22853] [Help] [answered]
When trying to read a file from `getServerSideProps` locally it works, but when deployed to Vercel, I get the following error:

```
2021-03-07T18:49:00.919Z	a9c0e291-7071-4b74-87ae-2b6bf3bc0359	ERROR	Error: ENOENT: no such file or directory, scandir '/vercel/workpath0/content/posts'
    at Object.readdirSync (fs.js:1021:3)
```

[Reproduction repo](https://github.com/robertcoopercode/with-mdx-remote-app)
Deployment URL: https://with-mdx-remote-app.vercel.app/

App structure

```
├─ pages
│  ├ posts.js
|    ├[slug].js
|  ├ index.js
│  └ rss.js
├─ posts
│  ├ example-1.js
│  └ example-2.js
└ next.config.js
```

In `rss.js` i'm trying to get the list of posts from the `posts` directory using [this strategy](https://github.com/vercel/next.js/issues/8251#issuecomment-547702744) to get the correct content path:

```js
export const getServerSideProps = (context) => {
  const res = context.res;
  const posts = fs.readdirSync(postsDirectory);
  console.log(posts);
  return { props: {} };
}
```

This works perfectly fine locally, but in a deployed environment, this does not work.

**Comment by robertcoopercode** [comment, marked as answer]
Nevermind, I've resolved the issue using `process.cwd()`.

**Reply by Babatunde50** [reply]
procss.cwd() works in development but not on production

**Reply by MiltonAkash** [reply]
Yes. It is working in local perfectly. But not on vercel

**Reply by MiltonAkash** [reply]
Works by adding `webpack5:false` in the next.config.js 

This may hurt the performance by a little bit. But I have no other way

**Comment by lumenwrites** [comment]
I'm struggling with the same issue, did you find the solution?

**Reply by MiltonAkash** [reply]
Add   `  webpack5: false`
 in your next.config.js

**Reply by morfeli** [reply]
This does not work anymore as webpack 4 is no longer supported in Next.js

**Reply by dowrow** [reply]
Same issue here

**Reply by wazcov** [reply]
same issue
